Testscripts/Linux/XDPForwardingTest.sh

```diff
@@ -0,0 +1,181 @@
+    exit 0
+}
+
+# Source constants file and initialize most common variables
+UtilsInit
+# Script start from here
+LogMsg "*********INFO: Script execution Started********"
+LogMsg "forwarder : ${forwarder}"
+LogMsg "receiver : ${receiver}"
+LogMsg "nicName: ${nicName}"
+bash ./XDPDumpSetup.sh ${forwarder} ${nicName}
+check_exit_status "XDPDumpSetup on ${forwarder}" "exit"
+SetTestStateRunning
+bash ./XDPDumpSetup.sh ${receiver} ${nicName}
+check_exit_status "XDpDUMPSetup on ${receiver}" "exit"
+SetTestStateRunning
+configure_XDPDUMP_TX
+
+LogMsg "XDP Setup Completed"
+
+# Setup pktgen on Sender
+LogMsg "Configure pktgen on ${sender}"
+pktgenDir=~/pktgen
+ssh ${sender} "mkdir -p ${pktgenDir}"
+download_pktgen_scripts ${sender} ${pktgenDir} ${cores}
+# Configure XDP_TX on Forwarder
+LogMsg "Build XDPDump with TX Action on ${forwarder}"
+ssh ${forwarder} "cd bpf-samples/xdpdump && make clean && CFLAGS='-D __TX_FWD__ -D __PERF__ -I../libbpf/src/root/usr/include' make"
+check_exit_status "Build xdpdump with TX Action on ${forwarder}"
+# Configure XDP_DROP on receiver
+LogMsg "Build XDPDump with DROP Action on ${receiver}"
+ssh ${receiver} "cd bpf-samples/xdpdump && make clean && CFLAGS='-D __PERF_DROP__ -D __PERF__ -I../libbpf/src/root/usr/include' make"
+check_exit_status "Build xdpdump with DROP Action on ${receiver}"
+
+# Calculate packet drops before tests
+packetDropBefore=$(ssh ${receiver} ". XDPUtils.sh && calculate_packets_drop ${nicName}")
+LogMsg "Before test, Packet drop count on ${receiver} is ${packetDropBefore}"
+# Calculate packets forwarded before tests
+pktForwardBefore=$(ssh ${forwarder} ". XDPUtils.sh && calculate_packets_forward ${nicName}")
+LogMsg "Before test, Packet forward count on ${forwarder} is ${pktForwardBefore}"
+
+# Start XDPDump on receiver
+xdpdumpCommand="cd bpf-samples/xdpdump && ./xdpdump -i ${nicName} > ~/xdpdumpout_${receiver}.txt"
+LogMsg "Starting xdpdump on ${receiver} with command: ${xdpdumpCommand}"
+ssh -f ${receiver} "sh -c '${xdpdumpCommand}'"
+# Start XDPDump on forwarder
+xdpdumpCommand="cd bpf-samples/xdpdump && ./xdpdump -i ${nicName} > ~/xdpdumpout_${forwarder}.txt"
+LogMsg "Starting xdpdump on ${forwarder} with command: ${xdpdumpCommand}"
+ssh -f ${forwarder} "sh -c '${xdpdumpCommand}'"
+
+# Start pktgen on Sender
+forwarderSecondMAC=$((ssh ${forwarder} "ip link show ${nicName}") | grep ether | awk '{print $2}')
+LogMsg "Forwarder second MAC: ${forwarderSecondMAC}"
+if [ "${core}" = "single" ];then
+    startCommand="cd ${pktgenDir} && ./pktgen_sample.sh -i ${nicName} -m ${forwarderSecondMAC} -d ${forwarderSecondIP} -v -n${packetCount}"
+    LogMsg "Starting pktgen on sender: $startCommand"
+    ssh ${sender} "modprobe pktgen; lsmod | grep pktgen"
+    result=$(ssh ${sender} "${startCommand}")
+else
+    startCommand="cd ${pktgenDir} && ./pktgen_sample.sh -i ${nicName} -m ${forwarderSecondMAC} -d ${forwarderSecondIP} -v -n${packetCount} -t8"
+    LogMsg "Starting pktgen on sender: ${startCommand}"
```

Comment:
dup code with above? it seems just different startcommand should be ok, all others following steps are the same in this simple if else fi.

Reply:
fixed